lib/saml_idp.rb

```diff
@@ -9,17 +9,26 @@ module SamlIdp
   require 'saml_idp/metadata_builder'
   require 'saml_idp/version'
   require 'saml_idp/engine' if defined?(::Rails) && Rails::VERSION::MAJOR > 2
+  require 'request_store'
 
   def self.config
-    @config ||= SamlIdp::Configurator.new
+    if RequestStore.store[:request_config]
+      RequestStore.store[:request_config]
+    else
+      raise "Config must be set before processing the request"
+    end
   end
 
   def self.configure
     yield config
   end
 
   def self.metadata
-    @metadata ||= MetadataBuilder.new(config)
+    MetadataBuilder.new(config)
+  end
+
+  def self.request_config=(config)
```

Comment:
Where is this used?

Reply:
This sets the per request config in host application.